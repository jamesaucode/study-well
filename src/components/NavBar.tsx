import React, { useState, useEffect } from "react";
import styled from "styled-components";
import Link from "next/link";
import ToggleableMenu from "./ToggleableMenu";
import { NextFC } from "next";
import { useLoginStatus } from "../hooks/useLoginStatus";
import { useWindowSize } from "../hooks/useWindowSize";
import BurgerMenu from "./BurgerMenu/BurgerMenu";

const NavWrapper = styled.div`
  width: 100%;
  padding: 1rem;
  box-sizing: border-box;
  border-bottom: 1px solid #eeeeee;
  display: flex;
  align-items: center;
`;
const Nav = styled.nav`
  display: flex;
  max-width: 1200px;
  /* height: 40px; */
  height: 100%;
  margin: 0 auto;
  align-items: center;
`;
export const NavLink = styled.li`
  font-family: Arial, Helvetica, sans-serif;
  text-transform: uppercase;
  margin-right: 4em;
  font-size: 0.9em;
  letter-spacing: 1px;
  list-style: none;
  font-weight: 600;
  color: #333333;
  cursor: pointer;
`;

const NavBar: NextFC = props => {
  const windowSize = useWindowSize();
  const [loading, setLoading] = useState<boolean>(true);
  const isLoggedIn = useLoginStatus();
  useEffect(() => {
    setLoading(false);
  }, [isLoggedIn]);
  if (loading) {
    return (
      <NavWrapper>
        <Nav />
      </NavWrapper>
    );
  } else if (windowSize > 700) {
    return (
      <NavWrapper>
        <Nav>
          <Link href="/">
            <NavLink>Home</NavLink>
          </Link>
          <Link href="/about">
            <NavLink>About</NavLink>
          </Link>
          {/* Protected Nav Items */}
          {isLoggedIn && (
            <React.Fragment>
              <Link href="/user/review">
                <NavLink>Review</NavLink>
              </Link>
              <Link href="/user/create">
                <NavLink>Make Card</NavLink>
              </Link>
              <Link href="/user/cards">
                <NavLink>Cards</NavLink>
              </Link>
            </React.Fragment>
          )}
        </Nav>
        <ToggleableMenu
          loggedIn={isLoggedIn}
          iconName="fas fa-user-circle"
          {...props}
        />
      </NavWrapper>
    );
  } else {
    return (
      <NavWrapper>
        <Nav>
          <BurgerMenu loggedIn={isLoggedIn} />
        </Nav>
      </NavWrapper>
    );
  }
};
export default NavBar;
