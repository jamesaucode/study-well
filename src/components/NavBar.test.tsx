/* eslint-disable */
import { mount } from "enzyme";
import NavBar from "./NavBar";

describe("NavBar", () => {
  it("Component renders", () => {
    const wrapper = mount(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
