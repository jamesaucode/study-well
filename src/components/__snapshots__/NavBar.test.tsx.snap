// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component renders 1`] = `
<NavBar>
  <NavBar__NavWrapper>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "NavBar__NavWrapper-xnd01y-0",
            "isStatic": true,
            "lastClassName": "degUwl",
            "rules": Array [
              "width:100%;height:100%;min-height:52px;box-sizing:border-box;box-shadow:0px 8px 18px rgba(0,0,0,0.05);display:flex;align-items:center;",
            ],
          },
          "displayName": "NavBar__NavWrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "NavBar__NavWrapper-xnd01y-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="NavBar__NavWrapper-xnd01y-0 degUwl"
      >
        <NavBar__Nav>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "NavBar__Nav-xnd01y-1",
                  "isStatic": true,
                  "lastClassName": "dOCoEg",
                  "rules": Array [
                    "align-items:center;display:flex;height:100%;width:100%;padding:0 1rem;max-width:700px;margin:0 auto;@media (max-width:600px){padding-left:0;padding-right:0;}",
                  ],
                },
                "displayName": "NavBar__Nav",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "NavBar__Nav-xnd01y-1",
                "target": "nav",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <nav
              className="NavBar__Nav-xnd01y-1 dOCoEg"
            >
              <Link
                href="/"
              >
                <NavBar__NavLink
                  onClick={[Function]}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "NavBar__NavLink-xnd01y-2",
                          "isStatic": true,
                          "lastClassName": "hLNIlQ",
                          "rules": Array [
                            "color:#333333;cursor:pointer;font-size:0.9em;font-weight:600;letter-spacing:1px;list-style:none;margin-right:4em;position:relative;text-transform:uppercase;&::before{background-color:",
                            "#0f76fc",
                            "80;bottom:-10px;content:\\"\\";height:4px;position:absolute;width:100%;transition:400ms ease transform;transform:scaleX(0);transform-origin:center;}&:hover:before{transform:scaleX(1);}",
                          ],
                        },
                        "displayName": "NavBar__NavLink",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "NavBar__NavLink-xnd01y-2",
                        "target": "li",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                  >
                    <li
                      className="NavBar__NavLink-xnd01y-2 hLNIlQ"
                      onClick={[Function]}
                    >
                      Home
                    </li>
                  </StyledComponent>
                </NavBar__NavLink>
              </Link>
            </nav>
          </StyledComponent>
        </NavBar__Nav>
        <ToggleableMenu
          loggedIn={false}
        >
          <ToggleableMenu__DropDownWrapper>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "ToggleableMenu__DropDownWrapper-sc-1qfmg6u-0",
                    "isStatic": true,
                    "lastClassName": "OPYsp",
                    "rules": Array [
                      "position:relative;display:inline-block;align-items:center;padding:0 1rem;margin:0 auto;",
                    ],
                  },
                  "displayName": "ToggleableMenu__DropDownWrapper",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "ToggleableMenu__DropDownWrapper-sc-1qfmg6u-0",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="ToggleableMenu__DropDownWrapper-sc-1qfmg6u-0 OPYsp"
              >
                <ToggleableMenu__ImportantLink
                  onClick={[Function]}
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "ToggleableMenu__ImportantLink-sc-1qfmg6u-4",
                          "isStatic": true,
                          "lastClassName": "bVfOgW",
                          "rules": Array [
                            "font-size:1em;font-weight:600;padding:0 1rem;color:",
                            "#ffffff",
                            "cc;text-decoration:none;",
                            "border:1px solid #0f76fc;border-radius:3px;color:#0f76fc;padding:0.5rem 1.5rem;font-size:1em;cursor:pointer;",
                          ],
                        },
                        "displayName": "ToggleableMenu__ImportantLink",
                        "foldedComponentIds": Array [
                          "ToggleableMenu__StyledLink-sc-1qfmg6u-3",
                        ],
                        "render": [Function],
                        "styledComponentId": "ToggleableMenu__ImportantLink-sc-1qfmg6u-4",
                        "target": "a",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    onClick={[Function]}
                  >
                    <a
                      className="ToggleableMenu__StyledLink-sc-1qfmg6u-3 ToggleableMenu__ImportantLink-sc-1qfmg6u-4 bVfOgW"
                      onClick={[Function]}
                    >
                      Login
                    </a>
                  </StyledComponent>
                </ToggleableMenu__ImportantLink>
              </div>
            </StyledComponent>
          </ToggleableMenu__DropDownWrapper>
        </ToggleableMenu>
      </div>
    </StyledComponent>
  </NavBar__NavWrapper>
</NavBar>
`;
